src/Extension/BetterNavigatorExtension.php

```diff
@@ -8,82 +8,157 @@
 use SilverStripe\Control\Director;
 use SilverStripe\Core\Config\Config;
 use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\FieldType\DBField;
 use SilverStripe\Security\LogoutForm;
 use SilverStripe\Security\Member;
 use SilverStripe\Security\Permission;
 use SilverStripe\Security\Security;
 use SilverStripe\Versioned\Versioned;
 use SilverStripe\View\ArrayData;
+use SilverStripe\ORM\FieldType\DBHTMLText;
+use SilverStripe\View\Requirements;
 
-class BetterNavigatorExtension extends DataExtension {
+class BetterNavigatorExtension extends DataExtension
+{
+
+    /**
+     * @var bool|null
+     */
+    private $shouldDisplay = null;
+
+    /**
+     * Load requirements in before final render. When the next extension point is called, it's too late.
+     * @return void
+     */
+    public function beforeCallActionHandler(): void
```

Comment:
I might be misunderstanding something but return type declarations are a php7 feature and SS4.2 only requires php5.6+ right - I think it's only SS4.5 onwards that will require php7? So if we're targeting 4.2 (or any currently released version) these declarations would have to go?

Reply:
Nah, 4.2 Is EOL as of 30 June. All new APIs should use PHP7 return type declarations. Not existing APIs, though! I've bumped the version constraint to reflect that.

Reply:
4.2 went EOL since the issue of this PR, so I've bumped it out of the dependency constraint.